enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,246 @@
+* Expose the index management API
+* Create and seed index templates to support the policy
+* Create the initial indices as needed
+* Block data ingestion if needed until initial index is seeded
+* Deploy curation CronJob for each mapping to rollover using the defined policy
+
+#### Collector 
+* Update the [`fluent-plugin-viaq-data-model`](https://github.com/ViaQ/fluent-plugin-viaq_data_model) to allow defining a static index to write logs
+
+Example configuration:
+```    
+  ....
+   <elasticsearch_index_name>
+     tag "**"
+     name_type static index_name=>'container.app-write'
+   </elasticsearch_index_name>
+```
+#### Deprecations
+* The Curator Cronjob deployed by the `cluster-logging-operator` will be removed. The responsibilities for curation will be subsumed by
+ implementation of Elasticsearch rollover management.
+* Curation configuration by namespace is no longer configurable and is restricted to cluster
+wide settings associated with log type 
+
+## Design Details
+
+### Test Plan
+* Regression tests will be executed to confirm no regressions from previous releases
+#### Unit Tests
+* Unit tests will be modified to account for the change in data design
+#### Integration Tests
+* e2e tests will be modified to account for the change in data design
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+The `elasticsearch-operator` will migrate existing indices to the new data design:
+* Indices beginning with `project.*` are indexed to `container.app-write`
+* Indices beginning with `.operations.*` are indexed to `node.infra-write`
+* Migrated indices are deleted after migration
+
+**Note:** Migrated container infrastructure logs are available via the `infra` alias only and
+subsequently curated from the cluster during period rollover.
+
+The `cluster-logging-operator` will remove previously deployed CronJobs that were owned by the ClusterLogging instance
+
+#### Downgrade
+Downgrades should be discouraged unless we know for certain the Elasticsearch version managed by cluster logging is the same version.  There is risk that Elasticsearch may have migrated data that is unreadable by an older version.
+
+## Implementation History
+
+| release|Description|
+|---|---|
+|4.4| GA release of rollover data design
+
+## Drawbacks
+
+The drawback to not implementing this change is that Cluster Logging will:
+* Continue to experience performance and scaling issues directly related to a less then optimal data schema
+
+## Alternatives
+
+There are no current alternatives
+
+
+## References
+
+* [Elasticsearch Rollover API documentation](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/indices-rollover-index.html)
+
+### Commands
+#### Create Template
+```
+curl http://localhost:9200/_template/app_logs?pretty -HContent-Type:application/json -XPUT -d '{
+  "index_patterns": ["app.container*"], 
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 1,
+  },
+  "aliases": {
+    "app.logs": {}
+   }
+}'
+```
```

Comment:
I just assume the `.all` alias will be still available, right?
It is useful for Kibana navigation when redirecting from OpenShift Web console, see [this Jira comment](https://jira.coreos.com/browse/LOG-452?focusedCommentId=149472&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-149472).

Reply:
Is this a public review of a design document that others outside of Red Hat can contribute to?  If so, then an internal Jira URL seems to be not as "open" as we should about this source.  If though, this is really just for Red Hat internal, then not sure why we are even tracking this document publicly (at least this repo appears to be public).

Reply:
> Is this a public review of a design document that others outside of Red Hat can contribute to? If so, then an internal Jira URL seems to be not as "open" as we should about this source. If though, this is really just for Red Hat internal, then not sure why we are even tracking this document publicly (at least this repo appears to be public).

This is a public forum to address and disseminate information.